test/mlir/krnl/entry_point.mlir

```diff
@@ -13,7 +11,7 @@ module {
 // CHECK:         llvm.mlir.global external constant @_entry_point_0_in_sig("[in_sig]\00")
 // CHECK:         llvm.mlir.global external constant @_entry_point_0_out_sig("[out_sig]\00")
 
-// CHECK-LABEL:   llvm.func @run_main_graph
+// CHECK-LABEL:   llvm.func @run_first_entry
```

Comment:
Shouldn't the default continue to be generated as `run_main_graph`?

Also, we should have at least one test to validate providing a different name.

Reply:
> Shouldn't the default continue to be generated as run_main_graph?

This is open for debate, IMO. I'm fine if the answer is "yes", but I'd like some clarification. 

With this change, if the user provides their own graph name, thus overriding `main_graph`, for an ONNX model, the entry point emitted would still be `run_main_graph`, somewhat defeating the purpose. I could update the behavior such that `run_main_graph` is emitted if the user didn't override the default graph name.

The other scenario is where an existing MLIR file is being compiled, where there is only one kernel entry point specified. In this case, the current behavior is to disregard the name of the entry point and always emit `run_main_graph`. TBH, this doesn't make sense to me, but if there are reasons/dependencies that rely upon this behavior, I can make the change more selective.

> Also, we should have at least one test to validate providing a different name.
I'll add an ONNX model graph test, then.

Reply:
So in the source, `main_graph` is still the default sometimes. There should be a test for this and if it's never the default, it should not be in the source. It sounds like we need at least 3 tests for the entry points.

Reply:
Updated the test to ensure the default scenario is tested and then a scenario where the single entry point has a non-default name.

Reply:
So the three tests I would expect to see are:
1. default "main_graph". My understanding is that if no name is specified, this is the name picked. Now both tests are specifying a name (though one of them is specifying "main_graph")
2. entry point with name in the graph
3. Using the command line option to specify a name other than "main_graph"

Reply:
Are there any existing tests that import a basic ONNX model? I see that the repository has a `mnist.onnx` in the `docs` directory, but AFAICT, there are no tests that act on an ONNX model.

Reply:
We don't have tests that import an ONNX model directly but @sorenlassen recenlty add tests that create a small ONNX model in JSON and pass it to `onnx-mlir` command line. One example is: https://github.com/onnx/onnx-mlir/blob/main/test/mlir/onnx/parse/test_abs.json. Hope it helps.